core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -81,19 +79,21 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
           return this._subtopics[i];
         }
       }
-      return false;
     };
 
     // The following function is only meant to be called from the undo of
     // a delete subtopic operation and not anywhere else. For adding subtopics
     // in general, use topic.addSubtopic()
-    Topic.prototype.undoDeleteSubtopic = function(id, title, skillIds) {
+    Topic.prototype.undoDeleteSubtopic = function(
+        id, title, skillIdsForSubtopic, isNewlyCreated) {
       var newSubtopic = SubtopicObjectFactory.createFromTitle(id, title);
-      for (var i = 0; i < skillIds.length; i++) {
-        newSubtopic.addSkillId(skillIds[i]);
+      for (var i = 0; i < skillIdsForSubtopic.length; i++) {
+        newSubtopic.addSkillId(skillIdsForSubtopic[i]);
       }
       this._subtopics.push(newSubtopic);
-      this._nextSubtopicId++;
+      if (isNewlyCreated) {
```

Comment:
Wouldn't you also need to update the IDs of subtopics that were newly created after the one that was subsequently deleted?

Reply:
Done, added IncrementId() and added tests for the same.